common/basic-auth/base/gatekeeper-deployment.yaml

```diff
@@ -0,0 +1,36 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: basic-auth
+  namespace: kubeflow
+spec:
+  replicas: 1
+  strategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: basic-auth
+    spec:
+      containers:
+      - args:
+        - --username=$(USERNAME)
+        - --pwhash=$(PASSWORDHASH)
+        command:
+        - /opt/kubeflow/gatekeeper
+        env:
+        - name: USERNAME
+          valueFrom:
+            secretKeyRef:
+              key: username
+              name: kubeflow-login
+        - name: PASSWORDHASH
+          valueFrom:
+            secretKeyRef:
+              key: passwordhash
+              name: kubeflow-login
+        image: gcr.io/kubeflow-images-public/gatekeeper:v20190211-v0.4.0-rc.1-119-g5098995b-e3b0c4
+        name: app
```

Comment:
i've been moving name to be the first member under containers since this is what kustomize uses as a key